near-contract-standards/src/non_fungible_token/core/core_impl.rs

```diff
@@ -88,6 +92,10 @@ impl NonFungibleToken {
             tokens_per_owner: enumeration_prefix.map(LookupMap::new),
             approvals_by_id,
             next_approval_id_by_id,
+            royalties: royalties_prefix.map(|prefix| {
+                // TODO: Figure out where the percent has to be coming from.
+                Royalties::new(prefix, 0)
```

Comment:
So, currently the `royalties.payout.percent` is a public field, and it seems that the expectation is that developers using this extension will adjust it. I find this contract-level percentage quite inflexible - 0% shared between all payouts is useless, 1-99% could be limiting or confusing, and 100% seems to be the only reasonable option, but then payout to the token owner needs to be explicitly listed (storage wasted).

In either cases, please, provide usage-level documentation on how to use this stuff.

Reply:
I was thinking that this could be done in a different way. Why don't we sum all the payouts and send the rest to the owner:

```
    pub fn create_payout(&self, balance: Balance, owner_id: &AccountId) -> Payout {
        let mut payout = Payout {
            payout: self
                .accounts
                .iter()
                .map(|(account, percent)| (account.clone(), apply_percent(percent, balance).into()))
                .collect(),
        };
        let rest = balance - payout.payout.values().fold(0, |acc, &sum| acc + sum);
        let owner_payout: u128 = payout.payout.get(owner_id).map_or(0, |x| x.0) + rest;
        payout.payout.insert(owner_id.clone(), owner_payout.into());
        payout
    }
```